Weather/src/app/factory/factoryClassTimeOfDay.ts

```diff
@@ -0,0 +1,34 @@
+import { TimeOfDayEnum } from "../enums/enumForTimeOfDay";
+
+export class factoryClassTimeOfDay {
+    private readonly currentHour: number;
+
+    constructor() {
+        this.currentHour = new Date().getHours();
+    }
+
+    setTimeOfDay(): TimeOfDayEnum {
+        if (this.currentHour >= 6 && this.currentHour < 12) {
+            return TimeOfDayEnum.MORNING;
+          } else if (this.currentHour >= 12 && this.currentHour < 18) {
+            return TimeOfDayEnum.DAY;
+          } else if (this.currentHour >= 18 && this.currentHour < 22) {
+            return TimeOfDayEnum.EVENING;
+          } else {
+            return TimeOfDayEnum.NIGHT;
+          }
+    }
+
+    createBackground(timeOfDay: TimeOfDayEnum): string {
+      switch (timeOfDay) {
+        case TimeOfDayEnum.MORNING:
+          return TimeOfDayEnum.MORNING;
+        case TimeOfDayEnum.DAY:
+          return TimeOfDayEnum.DAY;
+        case TimeOfDayEnum.EVENING:
+          return TimeOfDayEnum.EVENING;
+        case TimeOfDayEnum.NIGHT:
+          return TimeOfDayEnum.NIGHT;
+      }
+    }
```

Comment:
I think this method is useless)
You already have a value of class in TimeOfDayEnum.MORNING = 'morning-bg'. It's the same with method:
` createBackground(timeOfDay: TimeOfDayEnum): string { 
return timeOfDay;
}`